src/scripts/main.js

```diff
@@ -1,3 +1,34 @@
 'use strict';
 
 // write code here
+document.addEventListener('DOMContentLoaded', () => {
+  const headers = document.querySelectorAll('th');
+  const tableBody = document.querySelector('tbody');
+
+  const rows = Array.from(tableBody.querySelectorAll('tr'));
```

Comment:
The `rows` variable is defined outside the click event listener, which means it gets the initial state of the table rows. If the table gets updated dynamically (rows added or removed), this code won't work correctly. It would be better to move this line inside the click event listener to ensure it always gets the current state of the table rows.